src/index.ts

```diff
@@ -655,14 +655,21 @@ class DefaultMultiaddr implements Multiaddr {
 
   getPeerId (): string | null {
     try {
-      const tuples = this.stringTuples().filter((tuple) => {
-        if (tuple[0] === names.ipfs.code) {
-          return true
+      let tuples: Array<[number, string | undefined]> = []
+
+      this.stringTuples().forEach(([code, name]) => {
+        if (code === names.p2p.code) {
+          tuples.push([code, name])
+        }
+
+        // if this is a p2p-circuit address, return the target peer id if present
+        // not the peer id of the relay
+        if (code === names['p2p-circuit'].code) {
+          tuples = []
         }
```

Comment:
i find it confusing that we push to tuples and then empty tuples if p2p-circuit is found. 

Could this result in (eventually longer and currently unknown) multiaddr tuples being removed that shouldn't? Instead, should we just not call `tuples.push` if it's the p2p-circuit case for the current code?

Reply:
I don't think so, because circuit relay addresses are not supposed to be relayed themselves so whatever is after `p2p-circuit` should be what we are interested in.

We reset the tuples because whatever was before the `p2p-circuit` tuple is the relay address and we're trying to extract the PeerId of the target, not the relay.

It's trying to handle the case of:

`/ip4/foo/p2p/QmRelay/p2p-circuit` -> should return `null` not `QmRelay`